android/README.md

```diff
@@ -16,73 +25,79 @@ Build the demo module to see it in action, or just call this in `Activity`'s `on
-
-## Multiple envoy urls
-
-```java
-@Override
-protected void onCreate(Bundle savedInstanceState) {
-    super.onCreate(savedInstanceState);
-    // register to receive results
-    IntentFilter filter = new IntentFilter();
-    filter.addAction(BROADCAST_URL_VALIDATION_SUCCEEDED);
-    filter.addAction(BROADCAST_URL_VALIDATION_FAILED);
-    filter.addAction(ShadowsocksService.SHADOWSOCKS_SERVICE_BROADCAST); // see shadowsocks service behavior above
-    LocalBroadcastManager.getInstance(this).registerReceiver(mBroadcastReceiver, filter);
-
-    List<String> envoyUrls = Collections.unmodifiableList(Arrays.asList("https://allowed.example.com/path/", "socks5://127.0.0.1:1080"));
-    NetworkIntentService.submit(this, envoyUrls);
-
-    // or call NetworkIntentService.enqueueQuery, and
-    // we will get responses in Receiver's onReceive
-    // NetworkIntentService.enqueueQuery(this);
-    ...
-}
-
-protected final BroadcastReceiver mBroadcastReceiver = new BroadcastReceiver() {
-
-    @Override
-    public void onReceive(Context context, Intent intent) {
-        if (intent != null) {
-            if (intent.action == BROADCAST_URL_VALIDATION_SUCCEEDED) {
-                final List<String> validUrls = intent.getStringArrayListExtra(org.greatfire.envoy.NetworkIntentServiceKt..EXTENDED_DATA_VALID_URLS);
-                Log.i("BroadcastReceiver", "Received valid urls: " + TextUtils.join(", ", validUrls));
-                if (validUrls != null && !validUrls.isEmpty()) {
-                    String envoyUrl = validUrls.get(0);
-                    // the first listed url should be the fastest option
-                    // this will be triggered multiple times as additional urls are validated
-                    // consider adding code so that Cronet is only initialized once
-                    CronetNetworking.initializeCronetEngine(context, envoyUrl); // reInitializeIfNeeded set to false
+## Envoy url format
+
+Http/Https:
+ - http://domain:port (no trailing slash)
+ - https://domain:port (no trailing slash)
+
+Shadowsocks:
+ - ss://encrypted login@ip:port/
+
+Hysteria:
+ - hysteria://ip:port
+
+V2Ray Websocket:
+ - v2ws://domain:port:/id:uuid
+
+V2Ray SRTP:
+ - v2srtp://ip:port:uuid
+
+V2Ray WeChar:
+ - v2wechat://ip:port:uuid
+    
+## Submit envoy urls
+    
+There are two options for submitting envoy urls:
+    
+ - submit(context: Context, urls: List<String>)
+ - submit(context: Context, urls: List<String>, directUrl: String)
+    
+If the optional directUrl parameter is included, Envoy will attempt to connect to that url directly first.  This can be included to avoid using proxy resources when the target domain is not blocked.
+
+## Basic envoy integration
+
+```kotlin
+    private val mBroadcastReceiver: BroadcastReceiver = object : BroadcastReceiver() {
```

Comment:
why the mB style capitalization?

Reply:
it was part of the original example.  i can change it if you think it's important.